toolkit/tools/imagegen/diskutils/initramfs.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Utility to encrypt disks and partitions
+
+package diskutils
+
+import (
+	"bytes"
+	"io"
+	"os"
+
+	"github.com/cavaliercoder/go-cpio"
+	"github.com/klauspost/pgzip"
+	"microsoft.com/pkggen/internal/logger"
+)
+
+// InitramfsMount represented an editable initramfs
+type InitramfsMount struct {
+	pgzWriter           *pgzip.Writer
+	cpioWriter          *cpio.Writer
+	outputBuffer        *bytes.Buffer
+	initramfsOutputFile *os.File
+}
+
+// CreateInitramfs creates a new initramfs
+func CreateInitramfs(initramfsPath string) (initramfs InitramfsMount, err error) {
```

Comment:
>CreateInitramfs [](start = 5, length = 15)

The public functions are going to get used in the following two PRs, correct?

Reply:
Correct. I personally only use the OpenInitramfs() but I would like to see us using this common library for the ISO functionality as well which will need the Create function.

---
In reply to: [550321983](https://github.com/microsoft/CBL-Mariner/pull/481#discussion_r550321983) [](ancestors = 550321983)